README-xena-trafficgen-integration.md

```diff
@@ -0,0 +1,138 @@
+# README - Xena/trafficgen integration (using Valkyrie2544)
+
+## Installation & Setup
+
+1. For Linux, install Mono (https://www.mono-project.com/):
+* Notes:
+   * instructions are for CentOS/RHEL/Fedora, other distros please refer to https://www.mono-project.com/download/stable/#download-lin 
+   * commands should be run in a root shell or using `sudo`
+
+* Add Mono repo to system:
+```bash
+rpmkeys --import "http://pool.sks-keyservers.net/pks/lookup?op=get&search=0x3fa7e0328081bff6a14da29aa6a19b38d3d831ef"
+su -c 'curl https://download.mono-project.com/repo/centos8-stable.repo | tee /etc/yum.repos.d/mono-centos8-stable.repo'
+```
+
+   * Install Mono:
+```bash
+yum install mono-complete -y
+```
+
+2. Download the trafficgen git repository
+```bash
+git clone https://github.com/atheurer/trafficgen
+```
+
+
+3. Obtain .x2544 config file & executable
+   * Create a test config inside xena2544 utility on the xenaweb system (ask @ctrautma for IP if necessary)
+   * Save config into the ftp folder off the c drive
+   * Use FTP commands to download the config and the latest copy of Valkyrie2544.exe
+   * Place both config and exe inside the trafficgen folder
+```
+ftp 11.11.111.11 [user]/[pwd]
+bin
+get valkyrie2544.exe
+get <your config file>
+```
+
+5. Install TRex
+   * Within trafficgen, open `install-trex.sh` for editing
+   * Make the following change:
+        * original: `if curl --output ${tarfile} ${trex_url} && tar zxf ${tarfile}; then`
+        * modified: ``if curl --insecure --output ${tarfile} ${trex_url} && tar zxf ${tarfile}; then``
+   * As root or using `sudo`, run: `./install-trex.sh`
+   * After install completes, replace the following files with copies from test ZIP `fix_trex_install` (or make changes listed in Known Issues):
+        * `trex_client.py` --> `/opt/trex/current/automation/trex_control_plane/interactive/trex/common/trex_client.py`
+        * `trex_conn.py` --> `/opt/trex/current/automation/trex_control_plane/interactive/trex/common/trex_conn.py`
+        * `trex_global_stats.py` --> `/opt/trex/current/automation/trex_control_plane/interactive/trex/common/stats/trex_global_stats.py`
```

Comment:
The TRex inclusions are because of the following line from the beginning of binary-search.py:
`from trex_tg_lib import *`

Importing everything from trex_tg_lib in turn kicks off a chain of other imports - end result is that without TRex installed on a system, binary search crashes. For the sake of clarity I'll address the other questions in their own respective comments.

Reply:
Ok, that makes sense.  I'll have to think if there is a way to work around this.  Having to install TRex when you aren't using it just seems so broken...

Reply:
@ctrautma

I put together a patch that I think will solve these TRex dependency issues:

https://github.com/k-rister/trafficgen/commit/fa7235c1853061fa3e20c5f8ee7d6c1ac48d8e23

Can you have somebody test if this patch allows the Valkrie2544 stuff (this PR) to work without having the TRex stuff present as is described in the proposed README files?